### ISSUE TITLE
Is it possible to support Start Adornments and End Adornmentss?

### ISSUE THREAD

**Issue by abilash13** [TeamWertarbyte/material-ui-chip-input#248]
Right now, if we add a startAdornment and endAdornment to the InputProps, it looks and acts strange.

For example, this is how it looks for an empty chip input with start and end adornments:
<img width="505" alt="screenshot 2018-11-27 at 2 11 21 pm" src="https://user-images.githubusercontent.com/5894226/49069221-b736e880-f24e-11e8-8ec5-d63a62a28579.png">

This is how I need it to look:
<img width="497" alt="screenshot 2018-11-27 at 2 11 31 pm" src="https://user-images.githubusercontent.com/5894226/49069247-ccac1280-f24e-11e8-837b-33e1c712af20.png">

And when a new chip is added, this is how it looks:
<img width="496" alt="screenshot 2018-11-27 at 2 11 42 pm" src="https://user-images.githubusercontent.com/5894226/49069277-e2b9d300-f24e-11e8-9009-4f4669ce8171.png">

This is how I need it to look:
<img width="497" alt="screenshot 2018-11-27 at 2 16 39 pm" src="https://user-images.githubusercontent.com/5894226/49069366-1e549d00-f24f-11e8-851f-eb3db9036119.png">

Any idea if I can get this behavior by tweaking something?

**Comment by leMaik** [comment]
Indeed, it would be great if this worked! :open_mouth: 

@abilash13 If you want to give it a try, this looks like a styling issue. The `chipContainer`, `input` and `inputRoot` styles might be a good place to start (https://github.com/TeamWertarbyte/material-ui-chip-input/blob/master/src/ChipInput.js#L31).
I'd suggest to add a story with an input with adornments, so you could play around with hot reloading. :sunglasses:

**Comment by me245** [comment]
Is there any particular reason that most of the styling on the chip input is done with display: inline-block and margins instead of flex-box? Material-ui inputs are done with flex-box from what I am seeing. I am working on a project where I need this and due to time constraints am going to fix it inline. If there aren't already discussed reasons against flex-box I can make a pull request with my changes in order to get start adornments working and in the correct location

**Comment by leMaik** [comment]
@me245 No specific reason, might be some artifacts from the port from v0 to v1. Please feel free to change that, I'd be happy to see a PR.
Unfortunately, I'm _very_ busy until this christmas. :man_student:

**Comment by leMaik** [comment]
@me245 Any updates on this?

**Comment by me245** [comment]
I have a working solution for filled and outline inputs. I can make the changes for the default input too, just had a time constraint. As I am doing the default input, I am noticing a fair amount of duplicated css code that I am not liking. As I know other people are interested in this feature I can get that PR in, but I would want another issue put in to do some refactoring to use InputBase. I could use #244 as the issue to refactor, as that would be the fix for the issue that I am seeing

**Comment by ndtreviv** [comment]
I was trying it like this:
```
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <QuestionAnswerIcon />
                    </InputAdornment>
                  )
                }}
```

Which is the standard way of doing it with material-ui `TextField` objects. Obviously it doesn't work, so looking forward to this PR being merged!

**Comment by Mortuie** [comment]
Any progress on this? Would be great if it worked?

**Comment by sahanjith** [comment]
Any progress on this issue? Why the PR is not merged still?

**Comment by VGamezz19** [comment]
any update?

**Comment by ndtreviv** [comment]
Yeah, use this instead: https://material-ui.com/components/chips/
